lib/internal/webstreams/adapters.js

```diff
@@ -407,7 +413,12 @@ function newReadableStreamFromStreamReadable(streamReadable) {
   streamReadable.on('data', onData);
 
   return new ReadableStream({
-    start(c) { controller = c; },
+    start(c) {
+      controller = c;
+      if (!streamReadable._readableState.constructed) {
```

Comment:
I would just make start an async function here so that it always returns a promise.